src/include/function/gds/gds_object_manager.h

```diff
@@ -87,6 +90,15 @@ class AtomicObjectArray {
         }
         return false;
     }
+
+    void compare_and_swap(const common::offset_t pos, double valToAdd) {
```

Comment:
This is not what the function is doing. It's an addition using CAS, so call it addCAS.

Also for `compare_exchange_strong_max` do a similar naming with camel case.